common/component/azure/servicebus/client.go

```diff
@@ -121,13 +123,14 @@ func (c *Client) GetSender(ctx context.Context, queueOrTopic string, ensureFn en
 	}
 
 	// Ensure the queue or topic exists, if needed
-	if ensureFn != nil {
+	if ensureFn != nil && !c.topicCache[queueOrTopic] {
```

Comment:
Looks like this Azure component already has a cache. See above on L110 where it checks `c.senders` (btw this is a good example of how to use locks to prevent race conditions :) )

Reply:
lol how I dint look at that